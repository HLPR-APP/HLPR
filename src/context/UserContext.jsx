import { createContext, useContext, useState } from 'react';
import { getUser } from '../services/users';

const UserContext = createContext();

const UserProvider = ({ children }) => {
  const currentUser = getUser();
  const [user, setUser] = useState(
    currentUser ? { id: currentUser.id, email: currentUser.email } : null
  );
  return (
    <UserContext.Provider value={{ user, setUser }}>
      {children}
    </UserContext.Provider>
  );
};

const useUser = () => {
  const userCtx = useContext(UserContext);
  if (userCtx === undefined) {
    throw new Error('Must be wrapped in a UserProvider to use useUser');
  }
  return userCtx;
};

export { UserProvider, useUser };
