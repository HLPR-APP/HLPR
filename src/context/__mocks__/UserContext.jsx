import { createContext, useContext, useState } from 'react';

const UserContext = createContext();

const UserProvider = ({ mockUser, children }) => {
  const [user, setUser] = useState(
    mockUser ? { id: mockUser.id, email: mockUser.email } : null
  );
  return (
    <UserContext.Provider value={{ user, setUser }}>
      {children}
    </UserContext.Provider>
  );
};

const useUser = () => {
  const userCtx = useContext(UserContext);
  if (userCtx === undefined) {
    throw new Error('Must be wrapped in a UserProvider to use useUser');
  }
  return userCtx;
};

export { UserProvider, useUser };
