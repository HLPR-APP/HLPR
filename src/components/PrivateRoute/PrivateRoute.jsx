import { Redirect, Route } from 'react-router-dom';
import { useUser } from '../../context/UserContext';

export default function PrivateRoute({ children, ...rest }) {
  const { user } = useUser();
  return (
    <Route
      {...rest}
      render={({ location }) =>
        user ? (
          children
        ) : (
          <Redirect
            to={{
              pathname: '/',
              state: { from: location },
            }}
          />
        )
      }
    />
  );
}
