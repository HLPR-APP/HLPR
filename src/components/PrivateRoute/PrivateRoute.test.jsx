import React from 'react';
import dotenv from 'dotenv';
dotenv.config();
import { MemoryRouter } from 'react-router-dom';
import { render } from '@testing-library/react';
import PrivateRoute from './PrivateRoute.jsx';
import { UserProvider } from '../../context/UserContext.jsx';

jest.mock('../../services/users.js');
jest.mock('../../context/UserContext.jsx');

//need to define location because it is being used in render.
it('should render Private Route', () => {
  const { container } = render(
    <UserProvider>
      <MemoryRouter>
        <PrivateRoute location="/profile" />
      </MemoryRouter>
    </UserProvider>
  );
  expect(container).toMatchSnapshot();
});
