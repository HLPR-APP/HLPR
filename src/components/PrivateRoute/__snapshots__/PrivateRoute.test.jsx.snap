// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Private Route 1`] = `<div />`;
